src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/taglets/TagletWriter.java

```diff
@@ -372,10 +372,8 @@ public Content createAnchorAndSearchIndex(Element element, String tagText, Strin
     @SuppressWarnings("preview")
     Content createAnchorAndSearchIndex(Element element, String tagText, Content tagContent, String desc, DocTree tree) {
         Content result;
-        if (context.isFirstSentence && context.inSummary
-                || context.inTags.contains(DocTree.Kind.INDEX)
-                || htmlWriter instanceof IndexWriter
-                || htmlWriter instanceof SummaryListWriter<?>) {
+        if (context.isFirstSentence && context.inSummary || context.inTags.contains(DocTree.Kind.INDEX)
```

Comment:
Suggestion for bonus points:  how about introducing `context.isIndexable` to match `htmlWriter.isIndexable`